From 1eaabb5e05fed552bca014a9771a88b05702cd6b Mon Sep 17 00:00:00 2001
From: Shengzhou <shengzhou.liu@nxp.com>
Date: Mon, 3 Jun 2024 23:32:52 -0700
Subject: [PATCH] fix issue failed to boot kernel itb image

got boot errors as below on imx8 and imx93
Uncompressing Kernel Image
Error: inflate() returned -5
gzip compressed: uncompress error -1
Must RESET board to recover, resetting ...
---
 configs/imx8mm-cl-iot-gate-optee_defconfig                     | 2 +-
 configs/imx8mm-cl-iot-gate_defconfig                           | 2 +-
 configs/imx8mn_ab2_defconfig                                   | 2 +-
 configs/imx8mn_bsh_smm_s2_defconfig                            | 2 +-
 configs/imx8mn_bsh_smm_s2pro_defconfig                         | 2 +-
 configs/imx8mn_ddr3l_ab2_defconfig                             | 2 +-
 configs/imx8mn_ddr3l_evk_defconfig                             | 2 +-
 configs/imx8mn_ddr4_ab2_defconfig                              | 2 +-
 configs/imx8mn_ddr4_evk_android_defconfig                      | 2 +-
 configs/imx8mn_ddr4_evk_android_uuu_defconfig                  | 2 +-
 configs/imx8mn_ddr4_evk_defconfig                              | 2 +-
 configs/imx8mn_ddr4_evk_ld_defconfig                           | 2 +-
 configs/imx8mn_evk_android_defconfig                           | 2 +-
 configs/imx8mn_evk_android_dual_defconfig                      | 2 +-
 configs/imx8mn_evk_android_trusty_defconfig                    | 2 +-
 configs/imx8mn_evk_android_trusty_dual_defconfig               | 2 +-
 configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig | 2 +-
 configs/imx8mn_evk_android_uuu_defconfig                       | 2 +-
 configs/imx8mn_evk_defconfig                                   | 2 +-
 configs/imx8mn_evk_ld_defconfig                                | 2 +-
 configs/imx8mn_var_som_defconfig                               | 2 +-
 configs/imx8mp-icore-mx8mp-edimm2.2_defconfig                  | 2 +-
 configs/imx8mp_ddr4_evk_defconfig                              | 2 +-
 configs/imx8mp_ddr4_evk_inline_ecc_defconfig                   | 2 +-
 configs/imx8mp_ddr4_evk_nand_defconfig                         | 2 +-
 configs/imx8mp_evk_android_defconfig                           | 2 +-
 configs/imx8mp_evk_android_dual_defconfig                      | 2 +-
 configs/imx8mp_evk_android_powersave_defconfig                 | 2 +-
 configs/imx8mp_evk_android_trusty_defconfig                    | 2 +-
 configs/imx8mp_evk_android_trusty_dual_defconfig               | 2 +-
 configs/imx8mp_evk_android_trusty_powersave_dual_defconfig     | 2 +-
 configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig | 2 +-
 configs/imx8mp_evk_android_uuu_defconfig                       | 2 +-
 configs/imx8mp_evk_defconfig                                   | 2 +-
 configs/imx8mp_evk_inline_ecc_defconfig                        | 2 +-
 configs/imx8mp_evk_ndm_defconfig                               | 2 +-
 configs/imx8mp_rsb3720a1_4G_defconfig                          | 2 +-
 configs/imx8mp_rsb3720a1_6G_defconfig                          | 2 +-
 configs/imx8mq_cm_defconfig                                    | 2 +-
 configs/imx93_11x11_evk_defconfig                              | 2 ++
 40 files changed, 41 insertions(+), 39 deletions(-)

diff --git a/configs/imx8mm-cl-iot-gate-optee_defconfig b/configs/imx8mm-cl-iot-gate-optee_defconfig
index 524b22f3af..e5e0506166 100644
--- a/configs/imx8mm-cl-iot-gate-optee_defconfig
+++ b/configs/imx8mm-cl-iot-gate-optee_defconfig
@@ -42,7 +42,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_CMD_BOOTEFI_SELFTEST=y
 CONFIG_CMD_EXTENSION=y
 CONFIG_CMD_NVEDIT_EFI=y
diff --git a/configs/imx8mm-cl-iot-gate_defconfig b/configs/imx8mm-cl-iot-gate_defconfig
index 010fa91903..18c2a4bcf1 100644
--- a/configs/imx8mm-cl-iot-gate_defconfig
+++ b/configs/imx8mm-cl-iot-gate_defconfig
@@ -44,7 +44,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_CMD_BOOTEFI_SELFTEST=y
 CONFIG_CMD_EXTENSION=y
 CONFIG_CMD_NVEDIT_EFI=y
diff --git a/configs/imx8mn_ab2_defconfig b/configs/imx8mn_ab2_defconfig
index 0843878e78..3c150d9569 100644
--- a/configs/imx8mn_ab2_defconfig
+++ b/configs/imx8mn_ab2_defconfig
@@ -58,7 +58,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_bsh_smm_s2_defconfig b/configs/imx8mn_bsh_smm_s2_defconfig
index 8c0086fb1e..4b6e02b2cb 100644
--- a/configs/imx8mn_bsh_smm_s2_defconfig
+++ b/configs/imx8mn_bsh_smm_s2_defconfig
@@ -52,7 +52,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2067
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_CMD_FUSE=y
 CONFIG_CMD_USB=y
 CONFIG_CMD_USB_MASS_STORAGE=y
diff --git a/configs/imx8mn_bsh_smm_s2pro_defconfig b/configs/imx8mn_bsh_smm_s2pro_defconfig
index 57405d7935..c5778bd89e 100644
--- a/configs/imx8mn_bsh_smm_s2pro_defconfig
+++ b/configs/imx8mn_bsh_smm_s2pro_defconfig
@@ -48,7 +48,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2067
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_CMD_FUSE=y
 CONFIG_CMD_MMC=y
 CONFIG_CMD_USB=y
diff --git a/configs/imx8mn_ddr3l_ab2_defconfig b/configs/imx8mn_ddr3l_ab2_defconfig
index c235705a4c..cc4971a99e 100644
--- a/configs/imx8mn_ddr3l_ab2_defconfig
+++ b/configs/imx8mn_ddr3l_ab2_defconfig
@@ -59,7 +59,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_ddr3l_evk_defconfig b/configs/imx8mn_ddr3l_evk_defconfig
index 013282c201..40263c0789 100644
--- a/configs/imx8mn_ddr3l_evk_defconfig
+++ b/configs/imx8mn_ddr3l_evk_defconfig
@@ -59,7 +59,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_ddr4_ab2_defconfig b/configs/imx8mn_ddr4_ab2_defconfig
index b4263da91e..882581755c 100644
--- a/configs/imx8mn_ddr4_ab2_defconfig
+++ b/configs/imx8mn_ddr4_ab2_defconfig
@@ -54,7 +54,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="u-boot=> "
diff --git a/configs/imx8mn_ddr4_evk_android_defconfig b/configs/imx8mn_ddr4_evk_android_defconfig
index a205cda7f3..8423d4f6e8 100644
--- a/configs/imx8mn_ddr4_evk_android_defconfig
+++ b/configs/imx8mn_ddr4_evk_android_defconfig
@@ -55,7 +55,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="u-boot=> "
diff --git a/configs/imx8mn_ddr4_evk_android_uuu_defconfig b/configs/imx8mn_ddr4_evk_android_uuu_defconfig
index 856cd2229e..4fba83948e 100644
--- a/configs/imx8mn_ddr4_evk_android_uuu_defconfig
+++ b/configs/imx8mn_ddr4_evk_android_uuu_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="u-boot=> "
diff --git a/configs/imx8mn_ddr4_evk_defconfig b/configs/imx8mn_ddr4_evk_defconfig
index c316423a93..16f4f2f171 100644
--- a/configs/imx8mn_ddr4_evk_defconfig
+++ b/configs/imx8mn_ddr4_evk_defconfig
@@ -54,7 +54,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="u-boot=> "
diff --git a/configs/imx8mn_ddr4_evk_ld_defconfig b/configs/imx8mn_ddr4_evk_ld_defconfig
index 7e8fa0521b..f0ecca06b3 100644
--- a/configs/imx8mn_ddr4_evk_ld_defconfig
+++ b/configs/imx8mn_ddr4_evk_ld_defconfig
@@ -55,7 +55,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="u-boot=> "
diff --git a/configs/imx8mn_evk_android_defconfig b/configs/imx8mn_evk_android_defconfig
index d2a8f1bc8f..00eadd3c63 100644
--- a/configs/imx8mn_evk_android_defconfig
+++ b/configs/imx8mn_evk_android_defconfig
@@ -59,7 +59,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_evk_android_dual_defconfig b/configs/imx8mn_evk_android_dual_defconfig
index 2e0ca98dac..7821582466 100644
--- a/configs/imx8mn_evk_android_dual_defconfig
+++ b/configs/imx8mn_evk_android_dual_defconfig
@@ -59,7 +59,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_evk_android_trusty_defconfig b/configs/imx8mn_evk_android_trusty_defconfig
index d81e74b100..71c29ee879 100644
--- a/configs/imx8mn_evk_android_trusty_defconfig
+++ b/configs/imx8mn_evk_android_trusty_defconfig
@@ -59,7 +59,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_evk_android_trusty_dual_defconfig b/configs/imx8mn_evk_android_trusty_dual_defconfig
index f6ef5dcd2c..6b980187f2 100644
--- a/configs/imx8mn_evk_android_trusty_dual_defconfig
+++ b/configs/imx8mn_evk_android_trusty_dual_defconfig
@@ -59,7 +59,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig b/configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig
index 64202f6042..13695d2b5f 100644
--- a/configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig
+++ b/configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig
@@ -59,7 +59,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_evk_android_uuu_defconfig b/configs/imx8mn_evk_android_uuu_defconfig
index 3f48f18911..a3499ca985 100644
--- a/configs/imx8mn_evk_android_uuu_defconfig
+++ b/configs/imx8mn_evk_android_uuu_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_evk_defconfig b/configs/imx8mn_evk_defconfig
index df4e6243da..91f33c84a2 100644
--- a/configs/imx8mn_evk_defconfig
+++ b/configs/imx8mn_evk_defconfig
@@ -58,7 +58,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_evk_ld_defconfig b/configs/imx8mn_evk_ld_defconfig
index 2a6996d085..bfc3480241 100644
--- a/configs/imx8mn_evk_ld_defconfig
+++ b/configs/imx8mn_evk_ld_defconfig
@@ -59,7 +59,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_var_som_defconfig b/configs/imx8mn_var_som_defconfig
index f02cbab839..3de6a58086 100644
--- a/configs/imx8mn_var_som_defconfig
+++ b/configs/imx8mn_var_som_defconfig
@@ -48,7 +48,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2067
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_CMD_FUSE=y
 CONFIG_CMD_MMC=y
 CONFIG_CMD_USB=y
diff --git a/configs/imx8mp-icore-mx8mp-edimm2.2_defconfig b/configs/imx8mp-icore-mx8mp-edimm2.2_defconfig
index 24d288d541..cff381d55b 100644
--- a/configs/imx8mp-icore-mx8mp-edimm2.2_defconfig
+++ b/configs/imx8mp-icore-mx8mp-edimm2.2_defconfig
@@ -50,7 +50,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
 # CONFIG_CMD_CRC32 is not set
diff --git a/configs/imx8mp_ddr4_evk_defconfig b/configs/imx8mp_ddr4_evk_defconfig
index 83a4b958b0..a0d220edd0 100644
--- a/configs/imx8mp_ddr4_evk_defconfig
+++ b/configs/imx8mp_ddr4_evk_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=3
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_ddr4_evk_inline_ecc_defconfig b/configs/imx8mp_ddr4_evk_inline_ecc_defconfig
index 8463e86f13..e88f0202d3 100644
--- a/configs/imx8mp_ddr4_evk_inline_ecc_defconfig
+++ b/configs/imx8mp_ddr4_evk_inline_ecc_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=3
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_ddr4_evk_nand_defconfig b/configs/imx8mp_ddr4_evk_nand_defconfig
index 9c7e060eb8..1100a4a00b 100644
--- a/configs/imx8mp_ddr4_evk_nand_defconfig
+++ b/configs/imx8mp_ddr4_evk_nand_defconfig
@@ -55,7 +55,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=3
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_android_defconfig b/configs/imx8mp_evk_android_defconfig
index 322164fb35..02493db865 100644
--- a/configs/imx8mp_evk_android_defconfig
+++ b/configs/imx8mp_evk_android_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_android_dual_defconfig b/configs/imx8mp_evk_android_dual_defconfig
index 0bb84e91b7..a1943f5115 100644
--- a/configs/imx8mp_evk_android_dual_defconfig
+++ b/configs/imx8mp_evk_android_dual_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_android_powersave_defconfig b/configs/imx8mp_evk_android_powersave_defconfig
index 752965e1d7..56d5883c8e 100644
--- a/configs/imx8mp_evk_android_powersave_defconfig
+++ b/configs/imx8mp_evk_android_powersave_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_android_trusty_defconfig b/configs/imx8mp_evk_android_trusty_defconfig
index 0677c9256a..95c238eb3e 100644
--- a/configs/imx8mp_evk_android_trusty_defconfig
+++ b/configs/imx8mp_evk_android_trusty_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_android_trusty_dual_defconfig b/configs/imx8mp_evk_android_trusty_dual_defconfig
index 46523645b6..5ad34c53d0 100644
--- a/configs/imx8mp_evk_android_trusty_dual_defconfig
+++ b/configs/imx8mp_evk_android_trusty_dual_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_android_trusty_powersave_dual_defconfig b/configs/imx8mp_evk_android_trusty_powersave_dual_defconfig
index 6017f5c678..6ba33f8ca7 100644
--- a/configs/imx8mp_evk_android_trusty_powersave_dual_defconfig
+++ b/configs/imx8mp_evk_android_trusty_powersave_dual_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig b/configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig
index 4613d557fd..4068cb6413 100644
--- a/configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig
+++ b/configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_android_uuu_defconfig b/configs/imx8mp_evk_android_uuu_defconfig
index e34912c5a9..8bacfa0f12 100644
--- a/configs/imx8mp_evk_android_uuu_defconfig
+++ b/configs/imx8mp_evk_android_uuu_defconfig
@@ -61,7 +61,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_defconfig b/configs/imx8mp_evk_defconfig
index 658b63a455..fb23930d10 100644
--- a/configs/imx8mp_evk_defconfig
+++ b/configs/imx8mp_evk_defconfig
@@ -59,7 +59,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_inline_ecc_defconfig b/configs/imx8mp_evk_inline_ecc_defconfig
index 2c4448f55d..5894345a08 100644
--- a/configs/imx8mp_evk_inline_ecc_defconfig
+++ b/configs/imx8mp_evk_inline_ecc_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=3
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_ndm_defconfig b/configs/imx8mp_evk_ndm_defconfig
index c6917419b9..5381c39a04 100644
--- a/configs/imx8mp_evk_ndm_defconfig
+++ b/configs/imx8mp_evk_ndm_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_rsb3720a1_4G_defconfig b/configs/imx8mp_rsb3720a1_4G_defconfig
index 325242c540..325f447762 100644
--- a/configs/imx8mp_rsb3720a1_4G_defconfig
+++ b/configs/imx8mp_rsb3720a1_4G_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_CMD_BOOTEFI_SELFTEST=y
 CONFIG_CMD_ERASEENV=y
 CONFIG_CMD_NVEDIT_EFI=y
diff --git a/configs/imx8mp_rsb3720a1_6G_defconfig b/configs/imx8mp_rsb3720a1_6G_defconfig
index ede81fbd4c..2a757ffc16 100644
--- a/configs/imx8mp_rsb3720a1_6G_defconfig
+++ b/configs/imx8mp_rsb3720a1_6G_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_CMD_BOOTEFI_SELFTEST=y
 CONFIG_CMD_ERASEENV=y
 CONFIG_CMD_NVEDIT_EFI=y
diff --git a/configs/imx8mq_cm_defconfig b/configs/imx8mq_cm_defconfig
index 6db6003327..7117c50c80 100644
--- a/configs/imx8mq_cm_defconfig
+++ b/configs/imx8mq_cm_defconfig
@@ -44,7 +44,7 @@ CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR=0x300
 CONFIG_SPL_I2C=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_PBSIZE=1050
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_CMD_FUSE=y
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
diff --git a/configs/imx93_11x11_evk_defconfig b/configs/imx93_11x11_evk_defconfig
index ece2fcf1a5..ed66f43df9 100644
--- a/configs/imx93_11x11_evk_defconfig
+++ b/configs/imx93_11x11_evk_defconfig
@@ -27,7 +27,9 @@ CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_SYS_MONITOR_LEN=524288
 CONFIG_FIT=y
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_BOOTCOMMAND="run sr_ir_v2_cmd;run distro_bootcmd;run bsp_bootcmd"
-- 
2.34.1

